### ISSUE TITLE
Incompatible with poetry-core 1.6.0

### ISSUE THREAD

**Issue by layday** [mtkennerly/poetry-dynamic-versioning#121]
The internal `PyProjectTOML` class was changed in 1.6.0; it no longer has a `file` attribute:

```
[...]
  File "/tmp/build-env-3pbzw4dl/lib/python3.11/site-packages/poetry/core/masonry/api.py", line 66, in build_sdist
    poetry = Factory().create_poetry(Path(".").resolve(), with_groups=False)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/build-env-3pbzw4dl/lib/python3.11/site-packages/poetry_dynamic_versioning/patch.py", line 26, in alt_poetry_create
    pyproject_path=instance.pyproject.file,
                   ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'PyProjectTOML' object has no attribute 'file'
```

**Comment by mtkennerly** [comment]
Thanks for reporting this. I'll take a look.

**Comment by mtkennerly** [comment]
This is fixed now in v0.21.5.
